docs/DeveloperGuide.md

```diff
@@ -2,14 +2,219 @@
+This new account is then added to the list of accounts, and its number is added to the list of existing account numbers
+to ensure uniqueness.
+
+Finally, the method notifies the user of the successful addition of the account by displaying the details of the newly
+created account.
+
+The following sequence diagram shows how the add account process works:
+
+![](./uml/uml-images/addAccountDiagram.png)
+
+### [Implemented] Category feature
+
+#### Description
+
+The Category feature empowers users to effectively categorize transactions based on their preferences. When initiating a
+new transaction through the `Add` command, users are prompted to select a category from a predefined list. This ensures
+organized and streamlined transaction management.
+
+#### Design and Implementation
+
+The implementation of the Category feature revolves around the integration of a `category` attribute within each
+transaction object. This attribute is defined as a member of the `Category` enum class.
+
+Upon invoking the `Add` command, users are presented with a selection prompt featuring the available categories. User
+input, typically in the form of a numerical identifier corresponding to a category within the enum class, facilitates
+the assignment of the appropriate enum object to the transaction's category attribute.
+
+### [Implemented] Process transaction
+
+#### Description
+
+This method adds a transaction to the list of transactions based on the necessary input details given by the user.
+
+#### Parameters
+
+1. String input: A string containing the user input, which should include the `NAME`, `AMOUNT`, `DATE` and `TYPE` of the
+   transaction.
+2. Account account: The account object associated with the transaction list.
+
+#### Design and Implementation
+
+1. ##### Syntax Validation:
+
+   The method first checks whether the input string contains all necessary arguments ("/a/", "/t/", "/n/", "/$/", "/d/")
+   required for adding a transaction. If any argument is missing, it throws an InvalidAddTransactionSyntax exception.
+
+2. #### Transaction Parsing:
+
+   It utilizes a parser object to parse the user input string into a Transaction object using the
+   parseUserInputToTransaction method.
+
+3. #### Assertion Checks:
+
+   Assertions are used for debugging purposes to ensure that the parsed transaction and added transaction are not null.
+   If they are null, assertion errors are thrown.
+
+4. #### Category Assignment:
+
+   If the category of the transaction is null, it prompts the user to choose a category from a list and assigns the
+   chosen category to the transaction.
+
+5. #### Transaction Addition:
+
+   After parsing and category assignment, the transaction is added to the account using the addTransaction method.
+
+6. #### Feedback to User:
+
+   Upon successful addition of the transaction, a message is printed to the user indicating the details of the added
+   transaction and the updated account balance.
+
+#### Exceptions:
+
```

Comment:
clear diagram here. Would be good if we can highlight some exceptions here since this is in the Exceptions section, or should it be in another section?